src/pages/[productSlug]/integrations/[integrationSlug]/[integrationVersion]/components/[componentSlug]/index.tsx

```diff
@@ -0,0 +1,162 @@
+}
+
+async function _getServerSideProps({
+	params: { productSlug, integrationSlug, integrationVersion, componentSlug },
+}: {
+	params: PathParams
+}) {
+	// Pull out the Product Config
+	const product = cachedGetProductData(productSlug)
+
+	// 404 early if the product is not enabled for integrations
+	if (!ENABLED_INTEGRATION_PRODUCTS.includes(product.slug)) {
+		return {
+			notFound: true,
+		}
+	}
+
+	// Fetch the Integration
+	const integrationResponse = await fetchIntegration(
+		product.slug,
+		integrationSlug
+	)
+	if (integrationResponse.meta.status_code != 200) {
+		console.warn('Could not fetch Integration', integrationResponse)
+		return { notFound: true }
+	}
+	const integration = integrationResponse.result
+
+	// If the integration is external only, we shouldn't render this page
+	if (integration.external_only) {
+		return {
+			notFound: true,
+		}
+	}
+
+	// Fetch the Release
+	const activeReleaseResponse = await fetchIntegrationRelease(
+		product.slug,
+		integrationSlug,
+		integrationVersion === 'latest'
+			? integration.versions[0]
+			: integrationVersion
+	)
+	if (activeReleaseResponse.meta.status_code != 200) {
+		console.warn('Could not fetch Release', activeReleaseResponse)
+		return { notFound: true }
+	}
+	const activeRelease = activeReleaseResponse.result
+
+	// Grab the specific release component
+	const releaseComponent: ReleaseComponent = activeRelease.components
+		.filter((rc: ReleaseComponent) => {
+			// Only release components w/ either a README or variable groups get a page
+			return rc.readme || rc.variable_groups.length
+		})
+		.find((rc: ReleaseComponent) => {
+			return rc.component.slug === componentSlug
+		})
+	if (!releaseComponent) {
+		return { notFound: true }
+	}
+
+	// If the integration version is the latest version, redirect using `latest`
+	if (integrationVersion === integration.versions[0]) {
+		return {
+			redirect: {
+				destination: `/${product.slug}/integrations/${integration.slug}/latest/components/${releaseComponent.component.slug}`,
+				// Not permanent as a new release in the future will turn the
+				// latest release into an older release which should render!
+				permanent: false,
+			},
+		}
+	}
+
+	// Compute the string for the inclusion of the version in the title
+	const titleVersion =
+		activeRelease.version === integration.versions[0]
+			? ''
+			: ` (${activeRelease.version})`
+
+	return {
+		props: {
+			metadata: {
+				title: `${integration.name} ${releaseComponent.component.name}${titleVersion} | Integrations`,
+			},
+			product,
+			integration,
+			activeRelease,
+			component: releaseComponent,
+			serializedREADME: releaseComponent.readme
+				? await serializeIntegrationMarkdown(releaseComponent.readme)
+				: undefined,
+			breadcrumbLinks: integrationComponentBreadcrumbLinks(
+				product,
+				integration,
+				activeRelease,
+				releaseComponent,
+				true
+			),
+		},
+	}
+}
+
+export function integrationComponentBreadcrumbLinks(
```

Comment:
💭 No need to export this right?

```suggestion
function integrationComponentBreadcrumbLinks(
```

Reply:
Good callout, we should be safe here!